geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/stats/Stat.scala

```diff
@@ -102,6 +102,8 @@ trait Stat {
    * Necessary method used by the StatIterator.
    */
   def clear(): Unit
+
+  def newcopy: Stat
```

Comment:
could you add scala doc on what this is supposed to do? I assume that state should not be shared between the new copy and the existing stat... is existing state supposed to be copied over or is this just to clone an empty stat?